### ISSUE TITLE
None of the `hashes` fuzztests succeed

### ISSUE THREAD

**Issue by apoelstra** [rust-bitcoin/rust-bitcoin#1409]
We have fuzztests which try to produce a hash two ways and check that they're the same. Because our `cfg(fuzzing)` easily-broken hashes are implemented inconsistently, they don't pass.

```
RUSTFLAGS=--cfg=fuzzing cargo test
```

fails right now in `hashes/fuzz/`.

**Comment by tcharding** [comment]
Is the solution to check each failing test and see if it makes sense to run it when fuzzing, then just conditionally compile in the test only when not fuzzing? For example `sha256::tests::midstate` seems not useful when fuzzing. Or do I need to further understand what the fuzz tests do and what the hobbled code does when fuzzing?

**Comment by tcharding** [comment]
Now that #1821 merged exactly which cargo incantations should pass?

- In rust-bitcoin/hashes: `RUSTFLAGS=--cfg=fuzzing cargo test` -> PASS
- In rust-bitcoin/bitcoin: `RUSTFLAGS=--cfg=fuzzing cargo test` -> 12 tests FAIL
- In rust-bitcoin/: `RUSTFLAGS='--cfg=bitcoin_fuzz --cfg=hashes_fuzz --cfg=secp256k1_fuzz' cargo test` -> 85 tests FAIL

**Comment by tcharding** [comment]
This looks like its fixed,

- `cd fuzz;  RUSTFLAGS=--cfg=fuzzing cargo test` - all pass
- `cd bitcoin; RUSTFLAGS=--cfg=fuzzing cargo test` - all pass
- `cd hashes; RUSTFLAGS=--cfg=fuzzing cargo test` - all pass

**Comment by apoelstra** [comment]
Yeah, I seem to remember fixing this but I don't recall when exactly. The old fuzztests would compare against RustCrypto and I think we dropped those. Not sure if we replaced them with anything.

Ultimately there is very little value in fuzzing the hash crate at all. It doesn't allocate memory or index into arrays in complicated ways, and if the hashing code is wrong it will be wrong 100% of the time, so even a single unit test is enough.

Since the tests currently pass I'm happy to close this.
